connectors/filtering/basic_rule.py

```diff
@@ -94,6 +94,52 @@ def try_coerce(value):
     return coerced_value
 
 
+class RuleMatchStats:
+    def __init__(self, policy, matched_docs_count=0):
+        self.policy = policy
+        self.matched_docs_count = matched_docs_count
+
+    def increment_matched_doc_count(self, by=1):
```

Comment:
This is a very very long name for a method in Python. what about `incr_match` or `increment_match` ?  The general guideline is to try to keep names as small as possible. If you need a very long name to make it clear, it's usually a sign there's a design issue.

You can also look at `__add__`, if you implement it for `int` you could do:

```
rule_match = RuleMatchStats(policy)
rule_match += 1
```
but that could be a stretch

Reply:
Good point, implemented it 👍

Reply:
Also shortened some names, thanks for the review!